src/views/EditorView/EditorTopNavigationBar/EditorTopNavigationBar.tsx

```diff
@@ -0,0 +1,119 @@
+import {ContextType} from "../../../data/enums/ContextType";
+import './EditorTopNavigationBar.scss';
+import React from "react";
+import classNames from "classnames";
+import {AppState} from "../../../store";
+import {connect} from "react-redux";
+import {updateImageDragModeStatus} from "../../../store/general/actionCreators";
+import {GeneralSelector} from "../../../store/selectors/GeneralSelector";
+import {ViewPointSettings} from "../../../settings/ViewPointSettings";
+import {ImageButton} from "../../Common/ImageButton/ImageButton";
+import {ViewPortActions} from "../../../logic/actions/ViewPortActions";
+import {LabelsSelector} from "../../../store/selectors/LabelsSelector";
+import {LabelType} from "../../../data/enums/LabelType";
+import {AISelector} from "../../../store/selectors/AISelector";
+import {AIActions} from "../../../logic/actions/AIActions";
+
+interface IProps {
+    activeContext: ContextType;
+    updateImageDragModeStatus: (imageDragMode: boolean) => any;
+    imageDragMode: boolean;
+    activeLabelType: LabelType;
+}
+
+const EditorTopNavigationBar: React.FC<IProps> = ({activeContext, updateImageDragModeStatus, imageDragMode, activeLabelType}) => {
+
+    const getClassName = () => {
+        return classNames(
+            "EditorTopNavigationBar",
+            {
+                "with-context": activeContext === ContextType.EDITOR
+            }
+        );
+    };
+
+    const imageDragOnClick = () => {
+        if (imageDragMode) {
+            updateImageDragModeStatus(!imageDragMode);
+        }
+        else if (GeneralSelector.getZoom() !== ViewPointSettings.MIN_ZOOM) {
+            updateImageDragModeStatus(!imageDragMode);
+        }
+    };
+
+    return (
+        <div className={getClassName()}>
+            <div className="ButtonWrapper">
+                <ImageButton
+                    image={"ico/zoom-in.png"}
+                    imageAlt={"zoom-in"}
+                    buttonSize={{width: 30, height: 30}}
```

Comment:
widzę że wszystkie sizey są takie same :P zapisz to do zmiennej gdzieś powyżej będzie ładnie :D